spec/syntax.md

```diff
@@ -229,63 +210,55 @@ if it meets additional semantic requirements about its structure, defined below.
 
 ### Message
 
-A single message is either a plain message, a single pattern, or has a preamble
+A single message is either a single pattern, or has a `match` statement
 followed by one or more variants which represent the translatable body of the message.
 
 ```ebnf
-Message ::= Plain | Pattern | Preamble Variant+
+Message ::= Declaration* ( Pattern | Selector Variant+ )
 ```
 
-### Plain
+### Variable Declarations
 
-A plain message only contains translatable content;
-placeholders or their delimiters are not allowed inside a plain message.
-Plain messages must not start with one of the syntax characters `[`, `{` or `$`,
-as those would indicate that the message has a more complex structure.
-Any whitespace at the beginning or end of a plain message is ignored.
-A plain message cannot represent an empty string;
-for that, use an empty pattern `[]` instead.
+A variable declaration is an expression binding a variable identifier
+within the scope of the message to the value of an expression.
+This local variable may then be used in other expressions within the same message.
 
 ```ebnf
-Plain ::= PlainStart (PlainChar* PlainEnd)?  /* ws: explicit */
-PlainChar ::= AnyChar - ('{' | '}')
-PlainStart ::= PlainChar - ('[' | '$' | WhiteSpace)
-PlainEnd ::= PlainChar - WhiteSpace
+Declaration ::= 'let' WhiteSpace Variable '=' '{' Expression '}'
 ```
 
-### Preamble
+### Selectors
 
-The preamble is where selectors and local variables can be defined.
-A selector is an expression which will be used to choose one of the variants during formatting.
-A selector can be optionally bound to a local variable, which may then be used in other expressions.
+A selector is a statement containing one or more expressions
+which will be used to choose one of the variants during formatting.
 
 ```ebnf
-Preamble ::= Selector+
-Selector ::= (Variable '=')? '{' Expression '}'
+Selector ::= 'match' ( '{' Expression '}' )+
 ```
 
 Examples:
 
 ```
-$frac = {$count: number minFractionDigits=2}
-    1 [One apple]
-    * [{$frac} apples]
+let $frac = {$count: number minFractionDigits=2}
```

Comment:
How about using a simpler example here, without a local variable?

```
match {$count :plural}
when 1 [One apple]
when * [{$count} apples]
```

Reply:
Let's instead add a second example, so that we continue to cover the local variable declarations here a bit as well?

````suggestion
match {$count :plural}
when 1 [One apple]
when * [{$count} apples]
```

```
let $frac = {$count: number minFractionDigits=2}
````

Reply:
Sounds good, thanks.